src/cloud/aws/custom/paws.pm

```diff
@@ -839,6 +840,68 @@ sub directconnect_describe_virtual_interfaces {
     return $results;
 }
 
+sub cloudtrail_events {
+    my ($self, %options) = @_;
+
+    my $events_results = [];
+    eval {
+        my $ct;
+        if (length $self->{option_results}->{endpoint}) {
```

Comment:
it can be undefined. that syntax can do some undefined value with older perl version